payments-core/src/main/java/com/stripe/android/CreateIntentCallback.kt

```diff
@@ -0,0 +1,32 @@
+package com.stripe.android
+
+import androidx.annotation.RestrictTo
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun interface CreateIntentCallback {
+
+    suspend fun onCreateIntent(paymentMethodId: String): Result
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    sealed interface Result {
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        data class Success(val clientSecret: String) : Result
+
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        data class Failure(val error: Throwable) : Result
+    }
+}
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun interface CreateIntentCallbackForServerSideConfirmation : CreateIntentCallback {
+
+    override suspend fun onCreateIntent(paymentMethodId: String): CreateIntentCallback.Result {
+        error("This should not have been called")
```

Comment:
This means this relationship (`CreateIntentCallbackForServerSideConfirmation` implements `CreateIntentCallback`) is the wrong way to model this.

I'm fine with merging as is to unblock progress, but we should solve for this before GA.

Reply:
Yes, that’s what I meant above. We need to tweak the modelling here, but we want to make some progress on other fronts for now.